src/utils/getApplicationServerSideProps.ts

```diff
@@ -76,22 +81,31 @@ const redirectToLogin = (cookies, req, resolvedUrl) => {
   };
 };
 
+type ExtendedGetServerSidePropsContext = GetServerSidePropsContext & {
+  cookies: string;
+  queryClient: QueryClient;
+};
+
 const getApplicationServerSideProps =
-  (callbackFn) =>
-  async ({ req, query, resolvedUrl }) => {
+  (callbackFn?: (context: ExtendedGetServerSidePropsContext) => any) =>
+  async ({
+    req,
+    query,
+    resolvedUrl,
+    ...context
+  }: GetServerSidePropsContext) => {
     const { publicRuntimeConfig } = getConfig();
     if (publicRuntimeConfig.environment === 'development') {
-      process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
+      process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
     }
 
-    const rawCookies = req?.headers?.cookie ?? '';
-
-    const cookies = new Cookies(rawCookies, defaultCookieOptions);
+    const cookies = new Cookies(req.cookies);
```

Comment:
Aren't the `defaultCookieOptions` important?

Reply:
Typescript caught that this is actually not the place to pass them, this only accepts a custom cookie decoder but the options were ignored otherwise